### PR TITLE
feat: Support MXFP4 quantized dense models on AMD CDNA2/CDNA3 GPUs

### PR DESCRIPTION
**Pull request by fengli1702**
## Purpose

This PR adds support for MXFP4 quantized models on AMD CDNA2/CDNA3 GPUs through the Petit kernel backend. It enables vLLM to efficiently serve FP4-quantized models (e.g., models quantized by ModelOpt) on AMD GPUs that do not natively support FP4.

## Test Plan

Environment Setup
- Hardware: AMD MI300X GPUs with ROCm
- Models: amd--Llama-3.3-70B-Instruct (BF16 baseline) & amd--Llama-3.3-70B-Instruct-MXFP4-Preview
- Dependencies: vLLM with Petit kernel (petit-kernel >= 0.0.3)

**1. Smoke Test (Python API)**
```
python - <<'PY'
from vllm import LLM, SamplingParams

llm = LLM(
    model="/models",                  
    tensor_parallel_size=8,           
    quantization="petit_mxfp4",
    trust_remote_code=True,
    gpu_memory_utilization=0.9,
    enforce_eager=True,             
)

out = llm.generate(
    "San Francisco is a",
    SamplingParams(temperature=0, max_tokens=15),
)
print(out[0].outputs[0].text)
PY

```

## Test Result

Smoke Test generated_text: ' top destination for travelers, with its iconic Golden Gate Bridge, steep hills,'

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [x] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [x] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

File: vllm/model_executor/layers/quantization/petit.py [hunk 1]
```diff
@@ -317,3 +332,232 @@ def apply(
...
+            weight_loader=weight_loader,
+        )
+        layer.register_parameter("weight_scale_2", weight_scale_2)
+
+    def process_weights_after_loading(self, layer: torch.nn.Module) -> None:
+        prepare_mxfp4_layer_for_petit(layer)
+        global_scale = layer.weight_scale_2.max().to(torch.float32)
+        layer.weight_scale_2 = Parameter(
+            global_scale.reshape(1).to(layer.weight.device), requires_grad=False
+        )
```
[the remark this answers is not in the document]
Comment (fengli1702) [reply vllm/model_executor/layers/quantization/petit.py]: Thanks for the catch on fused-layer global scales: we kept the max(weight_scale_2) fallback because the Petit GEMM path currently accepts a single global scale, and I added an explicit warning_once when fused shards carry different weight_scale_2 values so the potential accuracy impact is visible.

File: vllm/model_executor/layers/quantization/utils/petit_utils.py [hunk 2]
```diff
@@ -65,46 +65,73 @@ def verify_petit_nvfp4_supported(quant_method: str, group_size: int | None) -> N
...
-    # 2. Call _import_petit_kernel() to trigger (or get) the import.
-    petit_kernel = _import_petit_kernel()
+def _check_petit_mxfp4_supported(
+    quant_method: str, group_size: int | None
+) -> tuple[bool, str | None]:
+    if quant_method != "MXFP4":
+        return (
+            False,
+            (
+                "Petit currently only supports: MXFP4 quantizations for this "
```
[the remark this answers is not in the document]
Comment (fengli1702) [reply vllm/model_executor/layers/quantization/utils/petit_utils.py:75]: Typo confirmed and fixed: the NVFP4 support error message in `petit_utils.py` now says “in vLLM”

File: vllm/model_executor/layers/quantization/petit.py [hunk 3]
```diff
@@ -317,3 +332,232 @@ def apply(
...
+    def get_min_capability(cls) -> int:
+        return 90
+
+    @classmethod
+    def get_config_filenames(cls) -> list[str]:
+        return ["hf_quant_config.json"]
+
+    @classmethod
+    def from_config(cls, config: dict[str, Any]) -> "PetitMxFp4Config":
+        qc = cls.get_from_keys(config, ["quantization"])
```
[the remark this answers is not in the document]
Comment (fengli1702) [reply vllm/model_executor/layers/quantization/petit.py]: Agreed on root-level config parsing: PetitMxFp4Config.from_config now accepts both nested quantization config and root-level quant_algo/quant_method, matching override_quantization_method behavior and preventing auto-detect initialization failures.
